CIP-015x/README.md

```diff
@@ -0,0 +1,404 @@
+
+Exact definition of the domain data is left out in this CIP. We propose that ledger team would propose canonical representation for the types in each new era. For the types they must be in a canonical CBOR format [cbor] with restrictions from [dcbor]. Values must not be derivable, that is, if some part of the state can be computed based on another part, then only the base one should be in the state."
+
+All concrete formats should be stored in attachment to this CIP and stored in `namespaces/namespaces.cddl`. All the changes should be introduced using current CIP update process.
+
+#### Canonicalization Rules
+
+- CBOR maps must be deterministic with sorted keys and no duplicates.
+- Numbers use minimal encoding.
+- Arrays follow fixed order.
+
+#### Verification
+
+- Entry digest: `digest(e) = H(0x01 || ns_str || key || value)`,
+- Manifest stores overall root and per-namespace commitments.
+
+Merkle root is computed as a root value of the Merkle trees over all the live entry digests in canonical order; tombstones excluded, last-writer-wins for overlays.
+
+To describe in detail, basic chunks store all the values in the canonically ordered based in the key order. After having all values in the order we build a full Merkle tree of those values.
+
+The rule of the thumb is that when we calculate a hash of the data we take into account only the live (non deleted) values in canonical order. In the case when there is a single dump without delta blocks, this is exactly the order of how values are stored. But when delta—blocks appear we need to take into account that in the following blocks there may be values that are smaller than the ones in the base and some values may be deleted or updated. As a result writer should calculate a live-set of values, which can be done by running a streaming multi-merge algorithm (when we search a minimal value from a multiple blocks). In the case a value exists in multiple blocks we use a last—writer—wins rule. If there is a tombstone, we consider a value deleted and do not include it in a live-set.
+
+### Extensibility
+
+- Unknown fields in `HDR` or unknown chunk types can be skipped by readers.
+- Allows future extension (e.g., index chunks, metadata) without breaking compatibility.
+
+## Rationale: how does this CIP achieve its goals?
+
+This CIP achieves its goals by:
+
+1. Define a canonical format that has the following properties: the format is very simple, it would be easy to write that an implementation in any languages.
+2. The format is extensible and agnostic, it provides versioned format that simplified ledger evolution.
+3. Removes ambiguity, allows signed checkpoints, and improves auditability
+
+Format defines canonical format and ordering for the stored data, thus allows reproducible and verifiable hashing of the data. It supports Mithril and fast node bootstrap use cases.
+
+### Prior Work and Alternatives
+
+#### Global alternatives:
+
+- **CIP PR #9 by Jean-Philippe Raynaud [jpraynaud-cip]**:
```

Comment:
There are a couple of different linking styles you could use here, with a "footnote" into the _References_ section being only one of the options... but currently this is not appearing as a link to the web site in the document footer....

Reply:
Oh.. sorry that part was supposed to be fixed, for some reason I didn't include it into PR. Will update ASAP, and look into other examples later as links may be nicer there

Reply:
I did another pass and fixed and rechecked all the links as it's done in CIP-0001.

Reply:
that's great... all links work now as well inline as they do in the References list. ⭐